### ISSUE TITLE
check-pipe totally broken

### ISSUE THREAD

**Issue by Martin-Wegner** [mgechev/codelyzer#365]
Since 3.1.2 check-pipe is totally broken.

```
<div *ngIf="countingPoints$ | async as countingPoints; else countingPoinsAreLoading">
  {{countingPoints}}
</div>
<ng-template #countingPoinsAreLoading>loading</ng-template>
```

> ERROR: XXX.component.html[2, 13]: The pipe operator should be surrounded by one space on each side, i.e. " | ".

**Comment by mgechev** [comment]
Thanks for reporting the issue. There are two issues in this case:

- The warning is reported for code having a valid style.
- The warning is reported on incorrect location.

I'll push a fix in upcoming releases.

**Comment by willgm** [comment]
Similar error here:

> ERROR: /home/willgm/proj/jexia-webapp/src/app/project/project-features/project-features.component.html[1, 25]: The property "async" that you're trying to access does not exist in the class declaration.

For this line:
```html
<div *ngIf="(currentProject | async).ui.features as features">
```

**Comment by mgechev** [comment]
@Martin-Wegner not a related issue. Please, take a look at the search.

**Comment by willgm** [comment]
@mgechev, sorry, but I can't see how this error is related to this issue. Can you explain?

I searched a little more, looks like there are other issues more likely to math the errors above: #190 and angular/angular-cli/issues/6554

This issue appeared after I updated from `3.0.1` to `3.1.0` or `3.0.2`:
```
~/proj/jexia-webapp [work ~2 *] » npm i -D codelyzer@3.1.0                              willgm@localhost
+ codelyzer@3.1.0
updated 1 package in 4.362s
---------------------------------------------------------------------
~/proj/jexia-webapp [work ~2 *] » npm run lint                                          willgm@localhost

> web-app@0.0.3-alpha lint /home/willgm/proj/jexia-webapp
> ng lint --type-check


ERROR: /home/willgm/proj/jexia-webapp/src/app/project/project-features/project-features.component.html[1, 2]: The property "async" that you're trying to access does not exist in the class declaration.
ERROR: /home/willgm/proj/jexia-webapp/src/app/project/project-features/project-features.component.html[55, 34]: The property "async" that you're trying to access does not exist in the class declaration.

Lint errors found in the listed files.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! web-app@0.0.3-alpha lint: `ng lint --type-check`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the web-app@0.0.3-alpha lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/willgm/.npm/_logs/2017-07-05T18_17_21_680Z-debug.log
---------------------------------------------------------------------
~/proj/jexia-webapp [work ~2 *] » npm i -D codelyzer@3.0.1                              willgm@localhost
+ codelyzer@3.0.1
updated 1 package in 4.294s
---------------------------------------------------------------------
~/proj/jexia-webapp [work ~2 *] » npm run lint                                          willgm@localhost

> web-app@0.0.3-alpha lint /home/willgm/proj/jexia-webapp
> ng lint --type-check



All files pass linting.
```

**Comment by wKoza** [comment]
Perharps, you should simply drop rule `no-access-missing-member` ?

**Comment by willgm** [comment]
@wKoza, now I got it, thanks! :tada:

**Comment by brian428** [comment]
Does anyone know if there's a way to disable this rule just for the templates that have the problem? I'd like to avoid just turning off the whole rule, since it only seems to fail in some limited situations (async pipe, inherited properties, etc.).

**Comment by mgechev** [comment]
You can disable the rule for the entire file by `/* tslint:disable:rule1 rule2 rule3... */`. In case the warning is in an external template, place the comment in the component file which references it.

**Comment by brian428** [comment]
Thanks, I was thinking it needed to go in the actual HTML somehow. I'll try it on the component itself.

**Comment by BorntraegerMarc** [comment]
Same thing in an HTML. This works:
```
<paper-item class="menu-item mi-new-task komed-menu-item" (click)="addNewTask()">{{ 'task.addTask' | translate }}</paper-item>
```

this generates an error:
```
<paper-item class="menu-item mi-new-task komed-menu-item" (click)="addNewTask()">
    {{ 'task.addTask' | translate }}
</paper-item>
```

this is the error: `The pipe operator should be surrounded by one space on each side, i.e. " | ".`
do you know when you can release a fix for this? My CI is broken because of this...

**Comment by mgechev** [comment]
You can disable the rule for now until we fix the issue.

**Comment by BorntraegerMarc** [comment]
How do I disable it?

**Comment by wKoza** [comment]
you should simply drop option `check-pipe` in your tslint file

**Comment by BorntraegerMarc** [comment]
@mgechev could you publish an update with this guy so we can get a clean CI build again? :)

**Comment by mgechev** [comment]
Yes, I'll do over the weekend. Will also do some dogfooding to verify that everything is fine.

**Comment by BorntraegerMarc** [comment]
@mgechev the issue seems not to be resolved for me:
```
D:\projects\komed-health-web>ng lint
ERROR: src/app/chat/chat.component.html[22, 43]: The pipe operator should be surrounded by one space on each side, i.e. " | ".

D:\projects\komed-health-web>ng --version
@angular/cli: 1.4.2
```

this is the HTML code:
```
<paper-item>
    {{ 'chat.switchToDe' | translate }}
</paper-item>
```

**Comment by wKoza** [comment]
Closed and reported to #416
